### PR TITLE
added notebooks for initial converting and input data checking

### PR DESCRIPTION
**Pull request by finn-freiheit**

### CODE REVIEW

File: 0_initial_converter.ipynb [hunk 1]
```diff
@@ -0,0 +1,158 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Initial Converter\n",
+    "\n",
+    "In this notebook we are converting the classification of the two measurement series\n",
+    "- 11th_tip3_pure3_500mV_12000pm_2000pA_5000ms_114\n",
```
Comment (vladhc) [inline 0_initial_converter.ipynb]: it would be convenient to have download links to the datasets

File: 0_initial_converter.ipynb [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+    "- 0_max_middle_500mV_8000pm_2000pA_5000ms_-5\n",
+    "into a classification file that obeys our specification.\n",
+    "\n",
+    "During the implementation of the project the specification might alter but currently (02.07.19) it is:\n",
```
Comment (vladhc) [inline 0_initial_converter.ipynb]: If it would alter - then this instructions would be updater, right? ;) Than no need for this intro text.
Also missing comma before "but".

File: 0_initial_converter.ipynb [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+    "import matplotlib.pyplot as plt"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "input_path = \"gitignored/\"\n",
```
Comment (vladhc) [inline 0_initial_converter.ipynb]: If I have a fresh install, how would these files appear in the "gitignored" directory? Wouldn't it be better to have some kind of installation script here, which would create the "dataset" directory, download and unzip data from the cloud storage?

File: 0_initial_converter.ipynb [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+    "\n",
+    "tip3_df = pd.read_csv(input_path + \"11th_tip3_pure3_500mV_12000pm_2000pA_5000ms_114_results.txt\", delimiter=\"\\t\", header=None, dtype=str)\n",
+    "tip3_df[0] = \"11th_tip3_pure3_500mV_12000pm_2000pA_5000ms_114_\" + tip3_df[0]\n",
```
Comment (vladhc) [inline 0_initial_converter.ipynb]: Would be more readable to reference the column by name instead of index. For that, probably, we need to provide column names, when reading the csv.

File: 0_initial_converter.ipynb [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+   "outputs": [],
+   "source": [
+    "hist_data = []\n",
+    "plot_bins = range(7)\n",
+    "\n",
+    "for i in plot_bins:\n",
+    "    hist_data.append(len(df[df[\"label\"] == i]))\n",
+    "\n",
+    "_ = plt.bar(plot_bins, hist_data)\n",
+    "plt.grid()"
```
Comment (vladhc) [inline 0_initial_converter.ipynb]: Would be nice to have axis names and title of the graph - plt.title() and plt.xlabel()/ylabel()

File: 0_initial_converter.ipynb [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "df[\"file_name\"] = df[\"file_name\"] + \".png\"\n",
+    "df.to_csv(input_path + \"classification.csv\", sep=\"\\t\")"
```
Comment (vladhc) [inline 0_initial_converter.ipynb]: we need somehow to skip clipping in certain cells. Or, maybe, embedding images here in the notebook? People should be able to see pretty graphs without installation of the Jupyter Lab. That would increase chances, that somebody will reuse the notebook.

File: 0_initial_converter.ipynb [hunk 7]
```diff
@@ -0,0 +1,158 @@
...
+   "outputs": [],
+   "source": [
+    "hist_data = []\n",
+    "plot_bins = range(6)\n",
+    "\n",
+    "for i in plot_bins:\n",
+    "    hist_data.append(len(df[df[\"label\"] == i]))\n",
+    "\n",
+    "_ = plt.bar(plot_bins, hist_data)\n",
+    "plt.grid()"
```
Comment (vladhc) [inline 0_initial_converter.ipynb]: The same: title of the graph, axis names.

File: 1_input_data_check.ipynb [hunk 8]
```diff
@@ -0,0 +1,190 @@
...
+   "metadata": {},
+   "source": [
+    "# Input Data Check\n",
+    "\n",
+    "In this notebook we will check the input data, display image samples and calculate statistics. This notebook and all the following expecting the input to be in the `gitignored/`-folder. The end goal is that the scientists only need to fill this folder and execute the notebooks or -- in a later phase -- the main program.\n",
+    "\n",
+    "As a member of the data science faction of freiheit.com, one can download an example data set with the following command:\n",
+    "\n",
+    "```\n",
+    "gsutil cp gs://stm-data-bode-europe-west1-b/example_data_set_1.zip .\n",
```
Comment (vladhc) [inline 1_input_data_check.ipynb]: If we are making this project public, I would ask freiheit.com to make this data accessible via public cloud storage. People should be able to easily reproduce our results. Without the data it's impossible. (Other way - make a web service, but as we discussed before, it's more effort).

File: 1_input_data_check.ipynb [hunk 9]
```diff
@@ -0,0 +1,190 @@
...
+    "df = pd.read_csv(input_folder + classification_file, delimiter=\"\\t\")\n",
+    "content_of_input_images_folder = os.listdir(input_images_folder)\n",
+    "\n",
+    "# check for missing files\n",
+    "missing_files = []\n",
+    "for file_name in df[\"file_name\"]:\n",
+    "    if file_name not in content_of_input_images_folder:\n",
+    "        missing_files.append(file_name)\n",
+    "     \n",
+    "if len(missing_files) != 0:\n",
```
Comment (vladhc) [inline 1_input_data_check.ipynb]: this should work also instead of these 3 lines:
assert len(missing_files) == 0, "There are missing files: {}".format(missing_files)
it's a bit more readable.

File: 1_input_data_check.ipynb [hunk 10]
```diff
@@ -0,0 +1,190 @@
...
+   "outputs": [],
+   "source": [
+    "hist_data = []\n",
+    "plot_bins = range(6)\n",
+    "\n",
+    "for i in plot_bins:\n",
+    "    hist_data.append(len(df[df[\"label\"] == i]))\n",
+    "\n",
+    "_ = plt.bar(plot_bins, hist_data)\n",
+    "plt.grid()\n",
```
Comment (vladhc) [inline 1_input_data_check.ipynb]: as before: plot title, axis names

File: 1_input_data_check.ipynb [hunk 11]
```diff
@@ -0,0 +1,190 @@
...
+    "In this notebook we will check the input data, display image samples and calculate statistics. This notebook and all the following expecting the input to be in the `gitignored/`-folder. The end goal is that the scientists only need to fill this folder and execute the notebooks or -- in a later phase -- the main program.\n",
+    "\n",
+    "As a member of the data science faction of freiheit.com, one can download an example data set with the following command:\n",
+    "\n",
+    "```\n",
+    "gsutil cp gs://stm-data-bode-europe-west1-b/example_data_set_1.zip .\n",
+    "unzip example_data_set_1.zip\n",
+    "```\n",
+    "\n",
+    "**Important**: We do not want to relabel images within our notebooks but we want to keep the same label mapping. When we train the neural network it should predict one of the six classes. But the classes must indexed from o on. So if we have six classes, they must be labeled `[0, 1, 2, 3, 4, 5]`. Therefore we just use this definition. So our `0` eqals to a `1` in [the paper]( https://pubs.acs.org/doi/abs/10.1021/acs.nanolett.7b02419)."
```
Comment (vladhc) [inline 1_input_data_check.ipynb]: that's really convenient to have a link to the paper hier!

File: 1_input_data_check.ipynb (hunk 8 above)
Comment (finn-freiheit) [reply 1_input_data_check.ipynb]: see here https://github.com/freiheit-com/deepphysics/issues/7

File: 0_initial_converter.ipynb (hunk 1 above)
Comment (finn-freiheit) [reply 0_initial_converter.ipynb]: I set up a script to download the data and create the `gitignored/` folder.

File: 0_initial_converter.ipynb (hunk 2 above)
Comment (finn-freiheit) [reply 0_initial_converter.ipynb]: removed

File: 0_initial_converter.ipynb (hunk 3 above)
Comment (finn-freiheit) [reply 0_initial_converter.ipynb]: I set up a script to download the data and create the `gitignored/` folder.

File: 0_initial_converter.ipynb (hunk 6 above)
Comment (finn-freiheit) [reply 0_initial_converter.ipynb]: issued here https://github.com/freiheit-com/deepphysics/issues/6

File: 0_initial_converter.ipynb [hunk 12]
```diff
@@ -0,0 +1,158 @@
...
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Initial Converter\n",
+    "\n",
+    "In this notebook we are converting the classification of the two measurement series\n",
+    "- 11th_tip3_pure3_500mV_12000pm_2000pA_5000ms_114\n",
+    "- 0_max_middle_500mV_8000pm_2000pA_5000ms_-5\n",
+    "into a classification file that obeys our specification.\n",
```
[the remark this answers is not in the document]
Comment (finn-freiheit) [reply 0_initial_converter.ipynb]: doone

File: 0_initial_converter.ipynb (hunk 4 above)
Comment (finn-freiheit) [reply 0_initial_converter.ipynb]: done :-) looks so much better!

File: 0_initial_converter.ipynb (hunk 5 above)
Comment (finn-freiheit) [reply 0_initial_converter.ipynb]: done

File: 0_initial_converter.ipynb (hunk 7 above)
Comment (finn-freiheit) [reply 0_initial_converter.ipynb]: done

File: 1_input_data_check.ipynb (hunk 9 above)
Comment (finn-freiheit) [reply 1_input_data_check.ipynb]: cool trick

File: 1_input_data_check.ipynb (hunk 10 above)
Comment (finn-freiheit) [reply 1_input_data_check.ipynb]: done
